### PR TITLE
fix: make upsert shop & merchant APIs RESTful + updated unit tests + minor refactors

### PR DESCRIPTION
**Pull request by onish-garg**

### CODE REVIEW

File: src/test/java/com/hyperlocal/server/MerchantControllerTest.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import static org.mockito.Mockito.when;
```
Comment (nagasrinivas-google) [inline src/test/java/com/hyperlocal/server/MerchantControllerTest.java:8]: You didn't have to make changes to ShopControllerTest class?

File: frontend/src/routes.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ const ROUTES = {
             'shopByShopID': base + '/v1/shops/%b'
         },
         'post':{
-            'insertMerchant': base+'/v1/insert/merchant'
```
Comment (nagasrinivas-google) [inline frontend/src/routes.js:22]: No url changes needed for ShopController in routes?

Comment (onish-garg) [reply frontend/src/routes.js:22]: No, all the screens dependent on `ShopController`'s routes are yet to be added.

File: src/test/java/com/hyperlocal/server/MerchantControllerTest.java (hunk 1 above)
Comment (onish-garg) [reply src/test/java/com/hyperlocal/server/MerchantControllerTest.java:8]: Short: No but now that I think of it, I should.
Long:  My understanding is, upsert merchant and upsert shop methods were both using helper functions to make calls to database. The unit tests were written specifically for these helper functions. In this PR, I changed unit tests for upsert merchant methods to test the actual methods being called and removed helper functions altogether; the helpers were small and not reused anywhere. I did not remove helper functions from upsert shop methods. On it right now.

Comment (nagasrinivas-google) [reply src/test/java/com/hyperlocal/server/MerchantControllerTest.java:8]: This file https://github.com/googleinterns/hyperlocal-gpay-spot/blob/master/src/test/java/com/hyperlocal/server/ShopControllerTest.java

**Pull request merged**